tests/test_components.py

```diff
@@ -537,6 +537,33 @@ def test_workflows(self):
             any_order=True,
         )
 
+    def test_parse_does_not_create_workflow_if_already_exists(self):
+        """
+        If the collected data already contains a given workflow, then it should be SELECTed, not CREATEd
+        """
+        user = UserFactory()
+        role = Role.objects.create(name="Role 1", user=user)
+        component = Workflows(Mock())
+        component.workflow(
+            "wf1",
+            {
+                "name": "Workflow 1",
+                "is_default": True,
+                "steps": [{"role": "role1", "is_required": True, "order": 1}],
+            },
+            {"role1": role},
+        )
+        component.workflow(
+            "wf2",
+            {
+                "name": "Workflow 1",
+                "is_default": True,
+                "steps": [{"role": "role1", "is_required": True, "order": 1}],
+            },
+            {"role1": role},
+        )
+        self.assertNotIn("wf2", component.data)
```

Comment:
You have me stumped, don't understand why "wf2" shouldn't be in component.data or how this assert checks anything in regards to `get_or_create`.
I mean, firstly, to even check that we're doing a db SELECT to get the existing workflow, we would have to actually have an existing workflow (i.e. create a workflow before we run `component.workflows()`).
Then in the assert part of the test you would have to check that no additional workflow was created and that no fields of the existing workflow have been changed (i.e. a db UPDATE has not run).
Only then do you know this is definitely working as expected.

Reply:
wf2 has the same name as wf1, so it should not be added.
Help me understand why this is not sufficient?